Feat15/schedule input

JSW0721 opened the pull request:
## 🔎 작업 내용

> 투약 스케쥴 입력 백엔드 작업
> 엔티티 수정

<br/>

## ➕ 이슈 링크

> #15 투약 스케쥴 생성

<br/>

## 이미지 첨부 (선택)

<br/>

JSW0721 committed:
브랜치 생성 커밋

JSW0721 committed:
개발 상황 중간 커밋

JSW0721 committed:
pull 충돌 병합 커밋

JSW0721 committed:
12/06작업 중간 커밋

JSW0721 committed:
12/06작업 커밋

JUNSIK2 commented on src/main/java/com/estsoft/projectdose/calendar/controller/CallenderController.java:
```diff
@@ -0,0 +1,76 @@
+		return "Calendar";
+	}
+
+	@RequestMapping("/CalendarList")
```
여기서 RequestMapping으로 사용한 이유가 있나요? 메소드 명시 안하면 모든 request가 이 메소드로 호출됩니다.

JUNSIK2 commented on src/main/java/com/estsoft/projectdose/calendar/controller/CallenderController.java:
```diff
@@ -0,0 +1,76 @@
+	}
+	//새로운 일정 추가
+	@PostMapping
+	public Calendar calendarSave(@RequestBody Map<String,Object> map) throws Exception{
```
변수 명을 map 보다는 정확한 용도에 명칭을 사용해 주는게 좋을 것 같습니다.

JUNSIK2 commented on src/main/java/com/estsoft/projectdose/calendar/controller/CallenderController.java:
```diff
@@ -0,0 +1,76 @@
+		return calendar;
+	}
+	@DeleteMapping("/CalendarDelete")
+	public String calendarDelete(@RequestParam Long id) throws Exception{
```
try catch 구문 사용하셨는데 throws 도 사용해야 하나요? 한번 확인해 주세요

JUNSIK2 commented on src/main/java/com/estsoft/projectdose/calendar/controller/CallenderController.java:
```diff
@@ -0,0 +1,76 @@
+		List<DoseScheduleViewResponse>doseschedule = cs.findAll().stream().map(DoseScheduleViewResponse::new).toList();
+
+		model.addAttribute("doseschedule", doseschedule);
+		return "Calendar";
```
여기는 페이지 리턴하는 메소드 같은데 @RestController로 사용하셨네요. 혹시 문제 있나 확인해주세요

icode1211 commented on src/main/java/com/estsoft/projectdose/calendar/controller/CallenderController.java:
```diff
@@ -59,14 +59,14 @@ public Calendar calendarSave(@RequestBody Map<String,Object> map) throws Excepti
 	public String calendarDelete(@RequestParam Long id) throws Exception{
 		try{
-			calendarservice.calendarDelete(id);
+			cs.calendarDelete(id);
```
CalendarService 객체는 줄임말 보다는 알아볼 수 있는 이름으로 사용하는게 더 나아보여요. (이전처럼)
라인수가 많아지다보면 cs 라는 변수명이 어떤것을 뜻하는지 알아보기가 어려울 수 있습니다

JUNSIK2 commented on src/main/java/com/estsoft/projectdose/calendar/entity/Calendar.java:
```diff
@@ -0,0 +1,27 @@
+@Getter
+@Setter
+@Entity
+public class Calendar {
```
혹시 db 테이블 변경 하셨나요? 수정한 부분 있으면 알려주세요

icode1211 commented on src/main/java/com/estsoft/projectdose/calendar/service/DoseScheduleService.java:
```diff
@@ -25,14 +25,15 @@ public DoseSchedule DoseScheduleSave(AddDoseScheduleRequest request){
-	public DoseSchedule findById(Long id) {
-		return dsr.deleteById(id);
+	public DoseSchedule findByid(Long id) {
+		return dsr.findById(id).orElse(null);
```
Optional을 리턴받았는데, 다시 null로 변환하는 것은 안티패턴 입니다! 다른 방법도 고려해보세요- 
Optional이 자바8 버전부터 생겨나게 된 이유를 생각해보시면 이해가 되실거에요 :)

icode1211 commented on src/main/java/com/estsoft/projectdose/calendar/entity/DoseSchedule.java:
```diff
@@ -36,5 +36,5 @@ public class DoseSchedule {
 
 	@Column(name = "is_deleted", nullable = false, columnDefinition = "BOOLEAN DEFAULT false")
-	private boolean isDeleted = false;
+	private boolean isDeleted;
```
잘 없애셨어요 👍

icode1211 commented:
구현하느라 고생하셨습니다. 
커밋 컨벤션도 맞춰주시면 좋을 것 같네요 
(간단 참고 : https://sungwookoo.tistory.com/1)

JSW0721 committed:
12/09작업 커밋

JSW0721 committed:
feat,test,chore
12월10일 작업 커밋. Calendar controller test 작성 및 JS,CSS,프론트엔드 작업 테스트 이후 Controller 수정작업

JSW0721 committed:
feat,test,chore
12월11일 작업 커밋. Calendar controller test 작성 및 JS,CSS,프론트엔드 작업 테스트 이후 Controller 수정작업

JSW0721 committed:
feat,chore
12월12일 작업 커밋. Calendar controller test 작성 및 JS,CSS,프론트엔드 작업 테스트 이후 Controller 수정작업, 불필요 엔티티,컨트롤러 제거

JSW0721 committed:
chore
12/12 develop브랜치 pull후 커밋

JSW0721 committed:
fix, chore
12/12 스케줄삭제 API명 수정, 불필요 테스트 삭제
